pkg/builder/builder.go

```diff
@@ -124,8 +128,13 @@ func (b *Builder) Prepare(cfgs ...interface{}) ([]string, error) {
 	warnings = append(warnings, isoWarnings...)
 	errs = packer.MultiErrorAppend(errs, isoErrs...)
 
-	if b.config.OutputDir == "" {
-		b.config.OutputDir = fmt.Sprintf("output-%s", b.config.PackerConfig.PackerBuildName)
+	if b.config.OutputDir != "" {
+		warnings = append(warnings, "output_directory is deprecated, use output_filename instead.")
+		b.config.OutputFile = fmt.Sprintf("output-%s/image", b.config.PackerConfig.PackerBuildName)
+	}
+
+	if b.config.OutputFile == "" {
+		b.config.OutputFile = fmt.Sprintf("output-%s/image", b.config.PackerConfig.PackerBuildName)
```

Comment:
i think the logic you here should be something like:
```
if b.config.OutputFile == "" {
 	if b.config.OutputDir != "" {
		warnings = append(...)
                 b.config.OutputFile = filepath.Join(b.config.OutputDir, "image")
	} else {
 		b.config.OutputFile = fmt.Sprintf("output-%s/image", b.config.PackerConfig.PackerBuildName) 
       } 
}
```

reading the current code, i think backwards compatibility will be broken:
if one provides OutputDir the image will be written to `output-{PackerBuildName}` and not to the output dir.

otherwise looks good!

Reply:
Ah you're right! I'll fix it tonight!

Reply:
Fixed